github3/repos/repo.py

```diff
@@ -2365,7 +2367,10 @@ def tree(self, sha):
         json = None
         if sha:
             url = self._build_url('git', 'trees', sha, base_url=self._api)
-            json = self._json(self._get(url), 200)
+            if recursive:
+                json = self._json(self._get(url, params={'recursive': 1}), 200)
+            else:
+                json = self._json(self._get(url), 200)
```

Comment:
Here I couldn't do
```python
params = {'recursive': 1 if recursive else 0}
json = self._json(self._get(url, params=params), 200)
```
because GitHub appears to only look for `recursive` in the query string, without caring for its value.

Reply:
What about:

```py
response = self._get(url, params=({'recursive': 1} if recursive else None))
json = self._json(response, 200)
```